public/app/features/explore/LogsNavigation.tsx

```diff
@@ -0,0 +1,243 @@
+      format: systemDateFormats.interval.second,
+      timeZone: timeZone,
+    })}`;
+  };
+
+  const oldestLogsFirst = logsSortOrder === LogsSortOrder.Ascending;
+  const theme = useTheme();
+  const styles = getStyles(theme, oldestLogsFirst, loading);
+  return (
+    <div className={styles.navContainer}>
+      {/*
+       * We are going to have 2 buttons - on the top and bottom - Oldest and Newest.
+       * Therefore I have at the moment duplicated the same code, but in the future iteration, it ill be updated
+       */}
+      {oldestLogsFirst && (
+        <Button
+          data-testid="fetchLogsTop"
+          className={styles.navButton}
+          variant="secondary"
+          onClick={() => {
+            // the range is based on initally selected range
+            changeTime({ from: visibleRange.from - requestRange, to: visibleRange.from });
+          }}
+          disabled={loading}
+        >
+          <div className={styles.navButtonContent}>
+            {loading ? <Spinner /> : <Icon name="angle-up" size="lg" />}
+            <div>Older</div>
+            <div>logs</div>
+          </div>
+        </Button>
+      )}
+      <CustomScrollbar autoHide>
+        <div className={styles.pagesWrapper}>
+          <div className={styles.pagesContainer}>
+            {pages.map((page: LogsPage, index) => (
+              <div
+                className={styles.page}
+                key={page.queryRange.to}
+                onClick={() => !loading && changeTime({ from: page.queryRange.from, to: page.queryRange.to })}
+              >
+                <div className={classNames(styles.line, { selectedBg: currentPageIndex === index })} />
+                <div className={classNames(styles.time, { selectedText: currentPageIndex === index })}>
+                  {`${formatTime(oldestLogsFirst ? page.logsRange.from : page.logsRange.to)} - ${formatTime(
+                    oldestLogsFirst ? page.logsRange.to : page.logsRange.from
+                  )}`}
+                </div>
+              </div>
+            ))}
+          </div>
+          <div className={styles.filler}></div>
+        </div>
+      </CustomScrollbar>
+
+      {!oldestLogsFirst && (
+        <Button
+          data-testid="fetchLogsBottom"
+          className={styles.navButton}
+          variant="secondary"
+          onClick={() => {
+            // the range is based on initally selected range
+            changeTime({ from: visibleRange.from - requestRange, to: visibleRange.from });
+          }}
+          disabled={loading}
+        >
+          <div className={styles.navButtonContent}>
+            <div>Older</div>
+            <div>logs</div>
+            {loading ? <Spinner /> : <Icon name="angle-down" size="lg" />}
+          </div>
+        </Button>
+      )}
+    </div>
+  );
+}
+
+export default memo(LogsNavigation);
+
+const getStyles = stylesFactory((theme: GrafanaTheme, oldestLogsFirst: boolean, loading: boolean) => {
```

Comment:
I'll work more on improving CSS in next iterations as I still need to add Newer button and Scroll to top button.